WcaOnRails/app/javascript/edit-events/ButtonActivatedModal.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import cn from 'classnames'
+import Modal from 'react-bootstrap/lib/Modal'
+import Button from 'react-bootstrap/lib/Button'
+import addEventListener from 'react-overlays/lib/utils/addEventListener';
+import ownerDocument from 'react-overlays/lib/utils/ownerDocument';
+
+export default class extends React.Component {
+  constructor() {
+    super();
+    this.state = { showModal: false };
+  }
+
+  open = () => {
+    this.setState({ showModal: true });
+  }
+
+  close = ({ skipUnsavedChangesCheck } = { skipUnsavedChangesCheck: false }) => {
+    if(!skipUnsavedChangesCheck && this.props.hasUnsavedChanges() && !confirm("Are you sure you want to discard your changes?")) {
+      return;
+    }
+
+    this.props.reset();
+    this.setState({ showModal: false });
```

Comment:
Ah, because I always default to writing my code by "bailing out" first. I'll flip this around as you suggest, because I know more people will find that normal.